### PR TITLE
11780: Emit event on invalid regexp value for aliasOrReg field

### PR DESCRIPTION
**Pull request by SamuelToh**
Pull request for [#11780](https://github.com/grafana/grafana/issues/11780) 

Please review the change, happy to resolve any potential issues. Thanks!

### CODE REVIEW

**Comment by torkelo** [comment]
need to update the test, some merge changes

File: public/app/core/time_series2.ts [hunk 1]
```diff
@@ -8,8 +9,12 @@ function matchSeriesOverride(aliasOrRegex, seriesAlias) {
   }
 
   if (aliasOrRegex[0] === '/') {
-    const regex = kbn.stringToJsRegex(aliasOrRegex);
-    return seriesAlias.match(regex) != null;
+    try {
+      const regex = kbn.stringToJsRegex(aliasOrRegex);
+      return seriesAlias.match(regex) != null;
+    } catch (e) {
+      return appEvents.emit('alert-error', ['Invalid aliasOrRegex value.', e.message]);
```
Comment (torkelo) [inline public/app/core/time_series2.ts]: Is there a way to thow an error here instead, don't want this low level model to issue an app level UI event

Comment (SamuelToh) [reply public/app/core/time_series2.ts]: Yep. Probably let the error bubble up to the (graph) panel level and handle it there. 

Thanks for the feedback. I'll have a go again tomorrow morning.

Comment (SamuelToh) [reply public/app/core/time_series2.ts]: @torkelo I have reworked the solution and squash them as 1 commit.

the lower level module `time_series2.ts` no longer catches the exception. Instead the error from the util lib is allowed to bubble up to the graph panel level where it is handled there. 

Please review and let me know if the change is alright.

Comment (torkelo) [reply public/app/core/time_series2.ts]: time_series 2 is also a low level model component that should not send messages

Comment (SamuelToh) [reply public/app/core/time_series2.ts]: ok thnks for the feedback will try again

Comment (SamuelToh) [reply public/app/core/time_series2.ts]: @torkelo Sorry for the late of response, was working on another PR and busy with work.

I have rework on the PR based on your recommendation. The `try-catch` block is now on the controller level. Hopefully it is ok now. Let me know if there is still issue with it. Happy to improve it.

**Comment by torkelo** [comment]
So merged this but the actual error handling still felt wrong. The error handling should be on the actual imput (so an error message to the right or below the series override input). 

The try catch should be on the whole render part and should show error state in the error corner of the panel. But tricky thing is when to clear the error state as there could be fetch (data) errors that should not be cleared in render.

**Pull request merged**